### DISCUSSION TITLE
how to delete Selected Records?

### DISCUSSION THREAD

**Discussion by ra-bedo11** [serenity-is/Serenity#6361] [Help] [answered]
![Selected_2](https://user-images.githubusercontent.com/29665710/162816488-15d0f953-b55f-462c-96d4-38ac098cdb71.png)

I Used This Code But not Work

In EndPoint

    public class DeleteMultiRequest : ServiceRequest
    {
        public List<int> Ids { get; set; }
    }

        [HttpPost, AuthorizeDelete(typeof(MyRow))]
        public ServiceResponse DeleteMulti(IUnitOfWork uow, DeleteMultiRequest request, [FromServices] ITblSocialStatusDeleteHandler handler)
        {
            request.Ids.ForEach(v =>
            {
                 handler.Delete(uow, new DeleteRequest() { EntityId = v });

                //return handler.Delete(uow, request);
            });

            return new ServiceResponse();
        }

and in Grid.ts

        private rowSelection: Serenity.GridRowSelectionMixin;

        protected createToolbarExtensions() {
            super.createToolbarExtensions();
            this.rowSelection = new Serenity.GridRowSelectionMixin(this);
        }

        protected getColumns(): Slick.Column[] {
            let columns = super.getColumns();
            let extOptions = this.get_ExtGridOptions();
            columns.splice(0, 0, Serenity.GridRowSelectionMixin.createSelectColumn(() => this.rowSelection));
            return columns;
        }

        protected getButtons() {
            let button = super.getButtons();
            button.push({
                title: "Delete Record",
                cssClass: "text-red",
                icon: "fa-times text-red",
                onClick: () => {
                    let selectedKeys = this.rowSelection.getSelectedKeys();
                    if (selectedKeys != null && typeof selectedKeys != 'undefined' && selectedKeys.length > 0) {
                        Q.confirm(
                            "are u sure delete ? ",
                            () => {
                                TblSocialStatusService.DeleteMulti({ Ids: selectedKeys() }, res => {
                                    Q.notifySuccess("Row is deleted");
                                    this.refresh();
                                });
                            },
                            {
                                onNo: () => {
                                    Q.notifyInfo("Row is not deleted");
                                },
                                onCancel: () => {
                                    Q.notifyError("You clicked X. Operation is cancelled. Will try again?");
                                },
                                dialogClass: 's-MessageDialog s-WarningDialog'
                            });
                    }

                    else {
                        Q.warning("Please select one row to delete");
                    }
                }
            })

            return button;
        }

![Selected_1](https://user-images.githubusercontent.com/29665710/162817233-69966acf-fe24-4a1f-b5a1-75f94d784614.png)

**Comment by wezmag** [comment]
The error message said "selectedKeys is not a function".
You need to remove `()` after selectedKeys in the line below.
```
TblSocialStatusService.DeleteMulti({ Ids: selectedKeys() }, res => {
```

**Reply by ra-bedo11** [reply]
When Remove This Line not work

**Comment by VictorTomaili** [comment, marked as answer]
`let selectedKeys = this.rowSelection.getSelectedKeys();` this code retrieves as string array. [Check this](https://github.com/serenity-is/Serenity/blob/66440d98d938032c943a0ce433923bc16000fabe/src/Serenity.Scripts/CoreLib/UI/Helpers/SlickHelpers.ts#L114=)
`TblSocialStatusService.DeleteMulti({ Ids: selectedKeys() }, res => ...` you cannot use this string array as a function. 
Your service may get the `Ids` as int array. This causes the type error. Check your service request types and pass them correctly.

**Reply by ra-bedo11** [reply]
thnks , its Work
